Update User Guide (#168)
AY2021S1-CS2103T-T11-1/tp: docs/UserGuide.md

* If optional parameter `[m/MONTH y/YEAR]` is present:
   * all sales whose associated date is in the specified `MONTH` and `YEAR` are listed.

**Example:** Let's say you want to view all sales made to Bernice Yu. This is what you need to do:

1. Type `contact find bernice` in the **Command Box** and press <kbd>Enter</kbd>. 
    - This is to identify the contact index corresponding to Bernice Yu.     
    
2. The **Result Box** will display a message of the number of contacts listed and the **Contact List** updates to
 show only contacts with the name 'Bernice'. Identify the correct index that corresponds to Bernice Yu, which is 1 in this example. 
 

> 3. Type `sale list c/1` in the **Command Box** and press <kbd>Enter</kbd>.\

Reviewer: Does this backslash affect anything?

Author: I've deleted the typo!

* Deletes the sale(s) at specified `SALE_INDEX`.
* The  `SALE_INDEX` refers to the index number shown in the displayed sale list, and must be a positive integer 1, 2, 3, …​
* Multiple `SALE_INDEX` can be specified, meaning that you can delete multiple sales with the same command.

**Examples:**
Let's say that you have just received the unfortunate news from Bernice Yu who no longer wishes to buy 2 drum sets from you.
Here's how you can delete this sale in StonksBook:

1. Type `contact find bernice` in the **Command Box** and press <kbd>Enter</kbd>. 
    - This is to identify the contact index corresponding to Bernice Yu.
    
2. The **Result Box** will display a message of the number of contacts listed and the **Contact List** updates to
 show only contacts with the name 'Bernice'. Identify the correct index that corresponds to Bernice Yu, which is 1 in this example.
 
3. Type `sale list c/1` in the **Command Box** and press <kbd>Enter</kbd>.

4. The **Result Box** will display a message noting that the command was successful, and the **Sale List** will
 show a list of sales made to Bernice Yu. Identify the index of the sale to be deleted. Let us assume it is at the third index.

> 5. Type `sale delete s/4` in the command box, and press Enter to execute it.

Reviewer: ```suggestion
5. Type `sale delete s/3` in the command box, and press <kbd>Enter</kbd> to execute it.
```

>     <img src="images/meeting-delete/meeting-delete.png" alt="Enter command 'meeting delete'" width="900px">

Reviewer: ```suggestion
```

* Edits the sale at the specified `SALE_INDEX`. 
* The `SALE_INDEX` refers to the index number shown in the displayed meeting list, and must be a positive integer 1, 2, 3, …​
* Multiple `SALE_INDEX` can be specified, meaning that you can edit multiple sales with the same command.
* At least one of the optional fields must be provided.
* Existing values will be updated to the input values.
* When editing a sale's tags, the existing tags of the sale will be removed i.e adding of tags is not cumulative.

**Examples:**
Suppose you have just received the good news that Bernice Yu would like to buy 20 Guitars instead of the 10 she previously specified. 
Here's how you can edit her sale in StonksBook:
 
1. Identify the index corresponding to the sale to be edited in the sale list. If you have a large list of
  sales, it may be convenient to filter for all sales made to Bernice Yu. Assuming that Bernice Yu is the 2nd contact
   currently displayed in the contact list, you can type `sale list c/2` and the sale list will update to show
    only sales made to Bernice Yu.
    

> 2. Suppose the sale to be edited is at the third index in the meeting list. Then, type `sale edit s/3 q/20` in the command box, and press <kbd>Enter</kbd> to execute it.

Reviewer: ```suggestion
2. Suppose the sale to be edited is at the third index in the sale list. Then, type `sale edit s/3 q/20` in the command box, and press <kbd>Enter</kbd> to execute it.
```

> 3. You should see that the meeting has been updated to reflect this new scheduled date.

Reviewer: ```suggestion
3. You should see that the sale has been updated to reflect this new quantity.
```

* Deletes the sale(s) at specified `SALE_INDEX`.
* The  `SALE_INDEX` refers to the index number shown in the displayed sale list, and must be a positive integer 1, 2, 3, …​
* Multiple `SALE_INDEX` can be specified, meaning that you can delete multiple sales with the same command.

**Examples:**
Let's say that you have just received the unfortunate news from Bernice Yu who no longer wishes to buy 2 drum sets from you.
Here's how you can delete this sale in StonksBook:

1. Type `contact find bernice` in the **Command Box** and press <kbd>Enter</kbd>. 
    - This is to identify the contact index corresponding to Bernice Yu.
    
2. The **Result Box** will display a message of the number of contacts listed and the **Contact List** updates to
 show only contacts with the name 'Bernice'. Identify the correct index that corresponds to Bernice Yu, which is 1 in this example.
 
3. Type `sale list c/1` in the **Command Box** and press <kbd>Enter</kbd>.

4. The **Result Box** will display a message noting that the command was successful, and the **Sale List** will
 show a list of sales made to Bernice Yu. Identify the index of the sale to be deleted. Let us assume it is at the third index.

> 5. Type `sale delete s/3` in the command box, and press <kbd>Enter</kbd> to execute it.

Reviewer: ```suggestion
5. Type `sale delete s/3` in the **Command Box**, and press <kbd>Enter</kbd> to execute it.
```

* Edits the sale at the specified `SALE_INDEX`. 
* The `SALE_INDEX` refers to the index number shown in the displayed meeting list, and must be a positive integer 1, 2, 3, …​
* Multiple `SALE_INDEX` can be specified, meaning that you can edit multiple sales with the same command.
* At least one of the optional fields must be provided.
* Existing values will be updated to the input values.
* When editing a sale's tags, the existing tags of the sale will be removed i.e adding of tags is not cumulative.

**Examples:**
Suppose you have just received the good news that Bernice Yu would like to buy 20 Guitars instead of the 10 she previously specified. 
Here's how you can edit her sale in StonksBook:
 
1. Identify the index corresponding to the sale to be edited in the sale list. If you have a large list of
  sales, it may be convenient to filter for all sales made to Bernice Yu. Assuming that Bernice Yu is the 2nd contact
   currently displayed in the contact list, you can type `sale list c/2` and the sale list will update to show
    only sales made to Bernice Yu.
    

> 2. Suppose the sale to be edited is at the third index in the sale list. Then, type `sale edit s/3 q/20` in the command box, and press <kbd>Enter</kbd> to execute it.

Reviewer: ```suggestion
2. Suppose the sale to be edited is at the third index in the sale list. Then, type `sale edit s/3 q/20` in the **Command Box**, and press <kbd>Enter</kbd> to execute it.
```
